### DISCUSSION TITLE
[Question][Hw4] About Worklist Algorithm

### DISCUSSION THREAD

**Discussion by SUNpluszero** [prosyslab-classroom/cs524-program-analysis#301] [Q&A]
Name: Seonyoung Jun

At the README of HW4, it says "Initially, only the entry point of main has the empty memory and all the other labels have the bottom memories (i.e., unreachable so far)."
I am not sure what is the difference between 'empty memory' and 'bottom memory'. And why do we need to distinguish them?

**Comment by KihongHeo** [comment]
Hi.

You can think in this way
- bot -> execution unreachable (because of nontermination, error, etc)
- empty -> reachable but no variables are defined
